omod/src/main/web/dashboardwidgets/obsacrossencounters/obsacrossencounters.controller.js

```diff
@@ -32,48 +35,78 @@ export default class ObsAcrossEncountersController {
        console.error(`failed to retrieve session info, error: ${error}`)
     });
   }
-  
-  fetchHeaders() {
-    if (this.config.headers) {
-      let columnNames = this.config.headers.split(",");
-      this.headers = columnNames;
-    }
-  }
 
   fetchConcepts() {
     let conceptKeys = this.config.concepts.split(",").map(c => c.trim());
-    for (let i = 0; i < conceptKeys.length; i++) {
-      this.openmrsRest.get("concept/" + conceptKeys[i], {
+    return Promise.all(conceptKeys.map((key) => (
+      this.openmrsRest.get("concept/" + key, {
         v: this.CONCEPT_CUSTOM_REP
       }).then((concept) => {
         // update the concept map with the REST representation of the concept
         concept.display = this.widgetsCommons.getConceptName(concept, "shortName", this.sessionLocale);
         this.conceptsMap[concept.uuid] = concept;
-      });
-    }
+      })
+    )));
   }
 
   fetchEncounters() {
     const encounterTypes = this.config.encounterTypes ? this.config.encounterTypes.split(',').map(c => c.trim()) : [];
     const legacyEncounterTypes = this.config.encounterType ? this.config.encounterType.split(',').map(c => c.trim()) : ["123AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"];
     encounterTypes.push(...legacyEncounterTypes);
 
-    const encounterPromises = encounterTypes.map(e =>
-      this.openmrsRest.get("encounter", {
-        patient: this.config.patientUuid,
-        encounterType: e,
-        v: 'custom:(uuid,encounterDatetime,encounterType:(name,description),obs:(id,uuid,value,concept:(id,uuid,name:(display),datatype:(uuid)),groupMembers:(id,uuid,display,value,concept:(id,uuid,name:(display),datatype:(uuid))))',
-        limit: this.config.maxRecords || 4,
-        fromdate: this.widgetsCommons.maxAgeToDate(this.config.maxAge),
-        order: this.order
```

Comment:
Did we lose this order parameter (maybe I'm just missing it) (and I'm not sure what it does)

Reply:
Oh, I see, it's the sort order, but it looks like it was hardcoded to "desc" so we just removed it?

Reply:
Yep. And sort was being handled client-side anyway.